stl/src/vector_algorithms.cpp

```diff
@@ -1978,6 +1970,116 @@ namespace {
         }
     }
 
+    struct _Count_traits_8 : _Find_traits_8 {
+#ifndef _M_ARM64EC
+        static __m256i _Sub_avx(const __m256i _Lhs, const __m256i _Rhs) noexcept {
+            return _mm256_sub_epi64(_Lhs, _Rhs);
+        }
+
+        static __m128i _Sub_sse(const __m128i _Lhs, const __m128i _Rhs) noexcept {
+            return _mm_sub_epi64(_Lhs, _Rhs);
+        }
+
+        static size_t _Reduce_avx(const __m256i _Val) noexcept {
+            const __m128i _Lo64 = _mm256_extracti128_si256(_Val, 0);
+            const __m128i _Hi64 = _mm256_extracti128_si256(_Val, 1);
+            const __m128i _Rx8  = _mm_add_epi64(_Lo64, _Hi64);
+#ifdef _M_IX86
+            return _mm_cvtsi128_si32(_Rx8) + _mm_extract_epi32(_Rx8, 2);
+#else // ^^^ defined(_M_IX86) / defined(_M_X64) vvv
+            return _mm_cvtsi128_si64(_Rx8) + _mm_extract_epi64(_Rx8, 1);
+#endif // ^^^ defined(_M_X64) ^^^
+        }
+
+        static size_t _Reduce_sse(const __m128i _Val) noexcept {
+#ifdef _M_IX86
+            return _mm_cvtsi128_si32(_Val) + _mm_extract_epi32(_Val, 2);
+#else // ^^^ defined(_M_IX86) / defined(_M_X64) vvv
+            return _mm_cvtsi128_si64(_Val) + _mm_extract_epi64(_Val, 1);
+#endif // ^^^ defined(_M_X64) ^^^
+        }

```

Comment:
I've checked the codegen that it is fine: https://godbolt.org/z/js9qoE5Mn.
It is fine.

Notice that the compiler encodes separate `_Reduce_sse` as `pextrq` / `movq`, and the inline `_Reuce_sse` as `vpextrq` / `vmovq`. It would have been a serious performance problem if it didn't use VEX-encoding for the inline version.

The compiler usually does the right thing, like in this case, but sometimes doesn't, see DevCom-10618264